code/game/objects/items/devices/doorCharge.dm

```diff
@@ -25,6 +25,13 @@
 			if(prob(25))
 				ex_act(1)
 
+/obj/item/device/doorCharge/Destroy()
+	if(istype(loc, /obj/machinery/door/airlock))
+		var/obj/machinery/door/airlock/A = loc
+		if(A.charge == src)
+			A.charge = null
+	return ..()
```

Comment:
so the reference to the doorcharge is cleared and the object isn't hard deleted anymore.